portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -12,15 +17,42 @@ public class FormHandlerServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+
+    if(!request.getParameter("text-input-name").matches("[\\w*\\s*]*")){
```

Comment:
You can improve code readability by adding a nethod called validateInput. Validate both name & email in the function. iIf there is an error, let the function return the error string. Return "" error string if successful.

Then your code can be simplified to
String error =  validateInput(request);
if (error.isEmpty()) {
  wrtieToDataStore()
  response.getWriter().println(someReturnString);
} else {
  response.getWriter().println(error);
}
do redirect